static/js/public/snap-details/videos.ts

```diff
@@ -29,35 +35,34 @@ function vimeo() {
 }
 
 function asciinema(
-  this: any,
-  holderEl: { querySelector: (arg0: string) => any }
-) {
+  holderEl: { querySelector: (arg0: string) => HTMLIFrameElement | null }
+): void {
   const asciinemaPlayer = holderEl.querySelector("iframe");
 
   if (!asciinemaPlayer) {
-    setTimeout(asciinema.bind(this, holderEl), 200);
+    setTimeout(() => asciinema(holderEl), 200);
     return;
   }
 }
 
-function videos(holderSelector: any) {
-  const holderEl = document.querySelector(holderSelector);
+function videos(holderSelector: string): void {
+  const holderEl = document.querySelector(holderSelector) as HTMLElement | null;
 
   if (!holderEl) {
     return;
   }
 
   const videoType = holderEl.dataset.videoType;
 
-  const iframe = holderEl.querySelector("iframe");
+  const iframe = holderEl.querySelector("iframe") as HTMLIFrameElement | null;
   if (iframe && iframe.src && iframe.src.indexOf("http://") !== -1) {
     iframe.src = iframe.src.replace("http://", "https://");
   }
 
   if (videoType === "vimeo") {
     vimeo();
   } else if (videoType === "asciinema") {
-    asciinema(holderEl);
+    asciinema(holderEl as HolderElement);
```

Comment:
Can this just be a built-in `HTMLElement`? Or do we need to be more explicit with the return type?

Reply:
Yes, I agree. It will simplify the type handling and I don't think we need any additional type constraints!